deps/wazuh_testing/wazuh_testing/data/qactl_conf_validator_schema.json

```diff
@@ -0,0 +1,367 @@
+                        "installation_files_path": {
+                          "type": "string"
+                        },
+                        "wazuh_install_path": {
+                          "type": "string",
+                          "default": "/var/ossec"
+                        }
+                      },
+                      "allOf": [
+                        {
+                          "if": {
+                            "properties": {"type": {"const": "sources"}}
+                          },
+                          "then": {"required": ["wazuh_branch"]},
+
+                          "else": {"required": ["local_package_path"]}
+                        },
+                        {
+                          "if": {
+                            "properties": {"target": {"const": "agent"}}
+                          },
+                          "then": {"required": ["manager_ip"]}       
+                        }        
+                      ]
+                    }
+                  }
+                },
+                "qa_framework": {
+                  "type": "object",
+                  "required": [
+                    "wazuh_qa_branch",
+                    "qa_workdir"
+                  ],
+                  "properties": {
+                    "wazuh_qa_branch": {
+                      "type": "string"
+                    },
+                    "qa_workdir": {
+                      "type": "string"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "tests": {
+      "type": "object",
+      "patternProperties": {
+        "^host[0-9]*$": {
+          "type": "object",
+          "required": ["host_info", "test"],
+          "properties": {
+            "host_info": {
+              "type": "object",
+              "required": [
+                "connection_method",
+                "host",
+                "user",
+                "connection_port",
+                "ansible_python_interpreter"
+              ],
+              "properties": {
+                "connection_method": {
+                  "type" : "string"
+                },
+                "host": {
+                  "type" : "string"
+                },
+                "user": {
+                  "type": "string"
+                },
+                "password": {
+                  "type": "string",
+                  "default": "empty"
+                },
+                "ssh_private_key_file_path":{
+                  "type": "string"
+                },
+                "connection_port": {
+                  "type": "integer"
+                },
+                "ansible_python_interpreter": {
+                  "type": "string"
+                }
+              },
+              "if": {
+                "properties": {"password": {"const": "empty"}}
+              },
+              "then": { "required": ["ssh_private_key_file_path"]}
```

Comment:
What happens if you enter both keys, would the result be valid?

Reply:
The result will be valid indeed as there isn't any type of exclusion added for validating just one parameter instead of both. For solving this problem, a `oneOf` schema implementation has been included, with this change, If both parameters are given, the validation will fail as it is required just one of them. This change can be viewed in the commit [cb1f8bf7ee0170d87f405b22ee031b065565f21d](https://github.com/wazuh/wazuh-qa/commit/cb1f8bf7ee0170d87f405b22ee031b065565f21d)